snakemake/executors/__init__.py

```diff
@@ -178,7 +184,28 @@ def print_job_error(self, job, msg=None, **kwargs):
         job.log_error(msg, **kwargs)
 
     def handle_job_success(self, job):
-        pass
+        if provenance_manager.is_active():
+            prov_mgr = provenance_manager
+            # print(job.params['biotools_id'])
+            tool_name = ""
+            if "biotools_id" in job.params.keys():
+                tool_name = job.params["biotools_id"]
+            else:
+                tool_name = job.__str__()
+            prov_mgr.add_activity(
+                tool_name=tool_name,
+                job_uri=job.uri,
+                input_id_list=job.input,
+                cmd=job.shellcmd,
+            )
+            outputs = job.output
+            for o in outputs:
+                prov_mgr.add_output(
+                    output_id=o,
+                    input_id_list=job.input,
+                    tool_name=tool_name,
+                    job_uri=job.uri,
+                )
```

Comment:
As you know, Snakemake has its own metadata tracking. I wonder if it would make sense to export the research object from there instead of doing this while running in here. The advantage is that you will get information also for stuff that happened in a previous run. The flag would then rather be a post-hoc command, just like --report, instead of requiring the user to remember to add it while running. Also, multiple partial runs of the same workflow do not result in several separate provenance information files.